Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+            // is to return (ideally an error would have been thrown before this point and the flow broken.)
+            guard let dataSource = indoorsLocationDataSource else { return }
+            locationDisplay.dataSource = dataSource
+            locationDisplay.autoPanMode = .compassNavigation
+            // Start the location display to zoom to the user's current location.
+            try await locationDisplay.dataSource.start()
+        }
+        
+        /// Creates an indoor location datasource from the maps tables if there is no indoors definition.
+        /// - Parameter map: The map which contains the tables from which the data source is constructed.
+        /// - Returns: Returns a configured IndoorsLocationDataSource created from the IPS position table.
+        private func createIndoorLocationDataSourceFromTables(map: Map) async throws -> IndoorsLocationDataSource? {
+            // Gets the positioning table from the map.
+            guard let positioningTable = map.tables.first(where: { $0.displayName == "IPS_Positioning" }) else { return nil }
+            // Creates and configures the query parameters.
+            let queryParameters = QueryParameters()
+            queryParameters.maxFeatures = 1
+            queryParameters.whereClause = "1 = 1"
+            // Queries positioning table to get the positioning ID.
+            let queryResult = try await positioningTable.queryFeatures(using: queryParameters)
+            guard let feature = queryResult.features().makeIterator().next() else { return nil }
+            let serviceFeatureTable = positioningTable as! ServiceFeatureTable
+            let positioningID = feature.attributes[serviceFeatureTable.globalIDField] as? UUID
+            
+            // Gets the pathways layer (optional for creating the IndoorsLocationDataSource).
+            let pathwaysLayer = map.operationalLayers.first(where: { $0.name == "Pathways" }) as! FeatureLayer
+            // Gets the levels layer (optional for creating the IndoorsLocationDataSource).
+            let levelsLayer = map.operationalLayers.first(where: { $0.name == "Levels" }) as! FeatureLayer
+            
+            // Setting up IndoorsLocationDataSource with positioning, pathways tables and positioning ID.
+            // positioningTable - the "IPS_Positioning" feature table from an IPS-aware map.
+            // pathwaysTable - An ArcGISFeatureTable that contains pathways as per the ArcGIS Indoors Information Model.
+            // Setting this property enables path snapping of locations provided by the IndoorsLocationDataSource.
+            // levelsTable - An ArcGISFeatureTable that contains floor levels in accordance with the ArcGIS Indoors Information Model.
+            // Providing this table enables the retrieval of a location's floor level ID.
+            // positioningID - an ID which identifies a specific row in the positioningTable that should be used for setting up IPS.
```

Comment:
This is essentially just duplicating the documentation for the parameters of `IndoorsLocationDataSource.init(positioningTable:pathwaysTable:levelsTable:positioningID:)`. I would suggest getting rid of it.

```suggestion
```